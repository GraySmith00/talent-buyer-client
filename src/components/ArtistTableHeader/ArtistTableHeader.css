.active {
  color: #CE2E7E; }
