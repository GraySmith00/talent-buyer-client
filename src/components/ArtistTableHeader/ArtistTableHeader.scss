@import '../../styles/variables.scss';

.active {
  color: $pink-three;
}
