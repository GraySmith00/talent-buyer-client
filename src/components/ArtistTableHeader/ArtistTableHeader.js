import React from 'react';
import './ArtistTableHeader.css';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';

const ArtistTableHeader = ({ category, onClick, activeSort, name }) => {
  const buttonClass = classNames({
    header: true,
    active: activeSort === name
  });
  return (
    <h3
      className={buttonClass}
      name={name}
      onClick={onClick}
      style={{ cursor: 'pointer' }}
    >
      {category}
    </h3>
  );
};

ArtistTableHeader.propTypes = {
  category: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  activeSort: PropTypes.string,
  name: PropTypes.string.isRequired
};

export default ArtistTableHeader;
