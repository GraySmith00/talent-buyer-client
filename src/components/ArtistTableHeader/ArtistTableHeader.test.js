import React from 'react';
import { shallow } from 'enzyme';
import { ArtistTableHeader } from './ArtistTableHeader';

describe('ArtistTableHeader', () => {
  let wrapper;

  beforeEach(() => {
    let mockCategory = 'mock';
    let mockFunc = jest.fn();
    let mockSort = 'mock';
    let mockName = 'mock';

    wrapper = shallow(
      <ArtistTableHeader
        category={mockCategory}
        onClick={mockFunc}
        activeSort={mockSort}
        name={mockName}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
