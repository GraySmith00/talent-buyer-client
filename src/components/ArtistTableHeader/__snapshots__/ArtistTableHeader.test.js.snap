// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistTableHeader should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    activeSort="mock"
    category="mock"
    name="mock"
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "mock",
      "className": "header active",
      "name": "mock",
      "onClick": [Function],
      "style": Object {
        "cursor": "pointer",
      },
    },
    "ref": null,
    "rendered": "mock",
    "type": "h3",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "mock",
        "className": "header active",
        "name": "mock",
        "onClick": [Function],
        "style": Object {
          "cursor": "pointer",
        },
      },
      "ref": null,
      "rendered": "mock",
      "type": "h3",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
